<launch>
  <arg name="mav_name" default="iris"/>
  <arg name="command_input" default="2" />
  <arg name="gazebo_simulation" default="false" />
  <arg name="max_acc" default="2.0" />
  <arg name="init_pos_x" default="0.0" />
  <arg name="init_pos_y" default="0.0" />
  <arg name="init_pos_z" default="2.0" />
  <arg name="pose_topic" default="/mavros/local_position/pose" />

  <node pkg="geometric_controller" type="geometric_controller_node" name="geometric_controller" output="screen">
  		<param name="mav_name" type="string" value="$(arg mav_name)" />
          <remap from="aft_mapped_pose" to="$(arg pose_topic)"/>
          <remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
          <param name="ctrl_mode" value="$(arg command_input)" />
          <param name="enable_sim" value="$(arg gazebo_simulation)" />
          <param name="enable_gazebo_state" value="true"/>
          <param name="velocity_yaw" value="false"/>
          <param name="max_acc" value="$(arg max_acc)" />
          <param name="attctrl_constant" value="0.45"/>
          <param name="normalizedthrust_constant" value="0.06"/>
          <param name="normalizedthrust_offset" value="0.1"/>
          <param name="Kp_x" value="4.0" />
          <param name="Kp_y" value="4.0" />
          <param name="Kp_z" value="13.5" />
          <param name="Kv_x" value="2" />
          <param name="Kv_y" value="2" />
          <param name="Kv_z" value="10" />
          <param name="init_pos_x" value="$(arg init_pos_x)" />
          <param name="init_pos_y" value="$(arg init_pos_y)" />
          <param name="init_pos_z" value="$(arg init_pos_z)" />
  </node>


  <!-- Launch rqt_reconfigure -->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" name="rqt_reconfigure" />
</launch>
